K2Bridge/Visitors/ElasticSearchDSLVisitor.cs

```diff
@@ -50,53 +50,63 @@ public void Visit(ElasticSearchDSL elasticSearchDSL)
 
-                queryStringBuilder.Append("| as aggs);");
-            }
+                    foreach (var (_, aggregation) in elasticSearchDSL.Aggregations)
+                    {
+                        aggregation.Accept(this);
+                        queryStringBuilder.Append($"{aggregation.KustoQL} ");
+                    }
 
-            // hits (projections...)
-            // The size is deserialized property
-            // therefore we check 'Size >= 0' to protect the query.
-            if (elasticSearchDSL.Size >= 0)
-            {
-                queryStringBuilder.Append("\n(_data ");
+                    queryStringBuilder.Append("| as aggs);");
+                }
 
-                if (elasticSearchDSL.Size > 0)
+                // hits (projections...)
+                // The size is deserialized property
+                // therefore we check 'Size >= 0' to protect the query.
+                if (elasticSearchDSL.Size >= 0)
                 {
-                    // we only need to sort if we're returning hits
-                    var orderingList = new List<string>();
+                    queryStringBuilder.Append("\n(_data ");
 
-                    foreach (var sortClause in elasticSearchDSL.Sort)
+                    if (elasticSearchDSL.Size > 0)
                     {
-                        sortClause.Accept(this);
-                        if (!string.IsNullOrEmpty(sortClause.KustoQL))
+                        // we only need to sort if we're returning hits
+                        var orderingList = new List<string>();
+
+                        foreach (var sortClause in elasticSearchDSL.Sort)
                         {
-                            orderingList.Add(sortClause.KustoQL);
+                            sortClause.Accept(this);
+                            if (!string.IsNullOrEmpty(sortClause.KustoQL))
+                            {
+                                orderingList.Add(sortClause.KustoQL);
+                            }
                         }
-                    }
 
-                    if (orderingList.Count > 0)
-                    {
-                        queryStringBuilder.Append($"| {KustoQLOperators.OrderBy} {string.Join(", ", orderingList)} ");
+                        if (orderingList.Count > 0)
+                        {
+                            queryStringBuilder.Append($"| {KustoQLOperators.OrderBy} {string.Join(", ", orderingList)} ");
+                        }
                     }
-                }
 
-                queryStringBuilder.Append($"| {KustoQLOperators.Limit} {elasticSearchDSL.Size} | as hits)");
+                    queryStringBuilder.Append($"| {KustoQLOperators.Limit} {elasticSearchDSL.Size} | as hits)");
+                }
+            }
+            else
+            {
+                // ViewSingleDocument query
+                queryStringBuilder.Append($"database(\"{databaseName}\").{tableName} {translatedQueryExpression} | as hits;");
```

Comment:
an easier way for this might be to always include the _data clause and here just append the condition with the id.

Reply:
I believe this will generate a messier query